src/Conversion/ONNXToKrnl/Math/Reduction.cpp

```diff
@@ -289,11 +289,94 @@ using MDBuilder =
 // one is needed, then pass ONNXNoneOp in the second slot.
 // Return true if we can optimize the reduction, false otherwise.
 
-// TODO: alexe add support for parallel
-// TODO: alexe see if the new simd infrastructure can be used.
+template <typename BUILDER, typename ONNXReductionOp1,
+    typename ONNXReductionOp2>
```

Comment:
It looks to me that we can generalize to multiple reduction ops. Same to `emitFullSIMDReductionFor`. But, it's ok for another PR.

Reply:
you are correct. Let's be on the lookout in models if we see opportunities.